### PR TITLE
feat: Add internal container accessor to `IFluidContainer`

### PR DESCRIPTION
**Pull request by Josmithr**
Adds an internal-only method to `IFluidContainer` for accessing its backing `IContainer`.

Needed to light up Client Debugger initialization from the 3rd party API surface without requiring creepy / brittle type-casts.

### CODE REVIEW

File: packages/framework/fluid-static/src/fluidContainer.ts [hunk 1]
```diff
@@ -187,6 +187,18 @@ export interface IFluidContainer extends IEventProvider<IFluidContainerEvents> {
...
+     * Gets the underlying {@link @fluidframework/container-definitions#IContainer}.
+     *
+     * FOR INTERNAL USE ONLY.
+     *
+     * @remarks Used to power debug tooling.
+     * @privateRemarks TODO: make this required in the future (breaking change).
+     *
+     * @internal
+     */
+    getInternalContainer?: () => IContainer;
```
Comment (Josmithr) [inline packages/framework/fluid-static/src/fluidContainer.ts:201]: @skylerjokiel Interested in your thoughts on naming here. I went with "InternalContainer" initially, but perhaps "InnerContainer" is better? Or something else?

**Review by skylerjokiel — CHANGES_REQUESTED** [review]

**Comment by Josmithr** [comment]
> This gives me some serious heartburn because it will get abused and we will be stuck supporting it in production scenarios. What are the alternative proposals?
>
> I'm thinking even having it on the FluidContainer itself and doing an any cast might be better since this will only be used for the debugger currently. We can always change it to a more supported api in the future if need be. Harder to take it away.

Per our offline discussion, I'll go ahead with moving this to only FluidContainer (and not the interface). I'm also going to add notes to the interface that we do not support external implementations which do not derive from FluidContainer, since I think it's important that we make that expectation clear.

**Comment by ChumpChief** [comment]
I think there are some other nice things we can do to prevent customers from taking dependencies on this - maybe let's chat today?

**Review by skylerjokiel — APPROVED** [review]
🚀  Looks good

**Pull request merged**